qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -464,13 +467,31 @@ def _format_data(self, data: CurveData) -> CurveData:
         Returns:
             Formatted CurveData instance.
         """
+        # take average over the same x value by keeping sigma
+        series, xdata, ydata, sigma, shots = multi_mean_xy_data(
+            series=data.data_index,
+            xdata=data.x,
+            ydata=data.y,
+            sigma=data.y_err,
+            shots=data.shots,
+            method="shots_weighted",
+        )
+
+        # sort by x value in ascending order
+        series, xdata, ydata, sigma = data_sort(
+            series=series,
+            xdata=xdata,
+            ydata=ydata,
+            sigma=sigma,
+        )
+
         return CurveData(
             label="fit_ready",
-            x=data.x,
-            y=data.y,
-            y_err=data.y_err,
-            data_index=data.data_index,
-            metadata=data.metadata,
+            x=xdata,
+            y=ydata,
+            y_err=sigma,
+            shots=shots,
+            data_index=series,
```

Comment:
What happened to the metadata?

Reply:
Currently it is only used by RB to compute EPG. Probably we need refactoring for this in another PR.

Reply:
Please make sure that this PR doesn't break computation of EPG for RB